veles/secrets/dockerhubpat/validator_test.go

```diff
@@ -0,0 +1,269 @@
+	t.Helper()
+
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Check if it's a POST request to the expected endpoint
+		if r.Method != http.MethodPost ||
+			r.URL.Path != "/v2/auth/token/" ||
+			r.Header.Get("Content-Type") != "application/json" {
+			t.Errorf("unexpected request: %s %s, expected: POST /v2/auth/token/ with content type application/json", r.Method, r.URL.Path)
+			http.Error(w, "not found", http.StatusNotFound)
+			return
+		}
+
+		// Check Body
+		bodyBytes, err := io.ReadAll(r.Body)
+		if err != nil {
+			t.Errorf("failed to read request body: %v", err)
+			http.Error(w, "internal server error", http.StatusInternalServerError)
+			return
+		}
+		defer r.Body.Close()
+		bodyString := string(bodyBytes)
+		expectedBody := fmt.Sprintf("{\"identifier\": \"%s\",\"secret\": \"%s\"}", expectedUser, expectedKey)
+		if expectedBody != bodyString {
+			w.WriteHeader(http.StatusUnauthorized)
+		}
+		w.WriteHeader(http.StatusOK)
+	}))
+}
+
+func TestValidator(t *testing.T) {
+	cases := []struct {
+		name        string
+		Username    string
+		Pat         string
+		want        veles.ValidationStatus
+		expectError bool
+	}{
+		{
+			name:     "invalid key and username",
+			Pat:      "dckr_oat_random",
+			Username: "User2",
+			want:     veles.ValidationInvalid,
+		},
+		{
+			name:     "valid key and username",
+			Pat:      validatorTestPat,
+			Username: validatorTestUsername,
+			want:     veles.ValidationValid,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Create a mock server
+			server := mockDockerHubServer(t, validatorTestPat, validatorTestUsername)
+			defer server.Close()
+
+			// Create a client with custom transport
+			client := &http.Client{
+				Transport: &mockTransport{testServer: server},
+			}
+
+			// Create a validator with a mock client
+			validator := dockerhubpat.NewValidator(
+				dockerhubpat.WithClient(client),
+			)
+
+			// Create a test username and pat
+			usernamePat := dockerhubpat.DockerHubPAT{Pat: tc.Pat, Username: tc.Username}
+
+			// Test validation
+			got, err := validator.Validate(context.Background(), usernamePat)
+
+			// Check error expectation
+			if tc.expectError {
+				if err == nil {
+					t.Errorf("Validate() expected error, got nil")
+				}
+			} else {
+				if err != nil {
+					t.Errorf("Validate() unexpected error: %v", err)
+				}
+			}
+
+			// Check validation status
+			if got != tc.want {
+				t.Errorf("Validate() = %v, want %v", got, tc.want)
+			}
+		})
+	}
+}
+
+func TestValidator_NoUsername(t *testing.T) {
+	testCases := []struct {
+		name     string
+		Pat      string
+		Username string
+		expected veles.ValidationStatus
+	}{
+		{
+			name:     "empty_username",
+			Pat:      validatorTestPat,
+			Username: "",
+			expected: veles.ValidationUnsupported,
+		},
+	}
+
+	for _, tc := range testCases {
```

Comment:
AFAICS this is the same logic as in the test above. Why not just include this as a separate testCase entry?